### PR TITLE
New default `geb` feature implementation

### PR DESCRIPTION
**Pull request by matrei**
- Rename feature `geb` to `geb-with-webdriver-binaries` and make it an optional feature
- Add new default feature `geb-with-testcontainers`

### CODE REVIEW

**Review by sbglasius — APPROVED** [review]
Looks good

**Comment by matrei** [comment]
The build is failing right now when running tests.
This is mimicking what the test is trying to do:
```console
mattias@mattias-z4-linux:/tmp/grailsforgetmp15664837821511079914$ ./gradlew build

FAILURE: Build failed with an exception.

* What went wrong:
Could not resolve all files for configuration ':buildSrc:buildScriptClasspath'.
> Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-ant:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-json:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-jmx:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-xml:4.0.23
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118 > org.apache.groovy:groovy-bom:4.0.23
   > Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
      > Could not get resource 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
         > Could not GET 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
            > Read timed out
> Could not resolve org.grails:grails-datastore-core:9.0.0-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-web:9.0.0-SNAPSHOT:20241025.200433-19
   > Skipped due to earlier error
> Could not resolve org.grails:grails-datastore-gorm:9.0.0-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-support:9.0.0-SNAPSHOT:20241025.200433-19
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-hibernate5:9.0.0-SNAPSHOT:20241026.013223-23
   > Skipped due to earlier error
> Could not resolve org.grails:grails-core:7.0.0-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails:grails-bom:7.0.0-SNAPSHOT:20241024.195419-118
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-support:9.0.0-SNAPSHOT:20241025.200433-19
   > Skipped due to earlier error
> Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
  Required by:
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-web:9.0.0-SNAPSHOT:20241025.200433-19
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-support:9.0.0-SNAPSHOT:20241025.200433-19
      project :buildSrc > org.grails.plugins:hibernate5:9.0.0-SNAPSHOT:20241026.013223-22 > org.grails:grails-datastore-gorm-hibernate5:9.0.0-SNAPSHOT:20241026.013223-23
   > Could not resolve org.apache.groovy:groovy:4.0.24-SNAPSHOT.
      > Could not get resource 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
         > Could not GET 'https://repo.grails.org/grails/core/org/apache/groovy/groovy/4.0.24-SNAPSHOT/groovy-4.0.24-SNAPSHOT.pom'.
            > Read timed out

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

BUILD FAILED in 2m 24s
1 actionable task: 1 up-to-date
```

File: grails-cli/src/test/groovy/org/grails/forge/cli/CommandSpec.groovy [hunk 1]
```diff
@@ -47,7 +47,7 @@ class CommandSpec extends Specification {
     Process executeGradleCommand(String command) {
         StringBuilder gradleCommand = new StringBuilder()
         if (spock.util.environment.OperatingSystem.current.isWindows()) {
-            gradleCommand.append("gradlew.bat")
+            gradleCommand.append("cmd.exe /c gradlew")
```
Comment (jamesfredley) [inline grails-cli/src/test/groovy/org/grails/forge/cli/CommandSpec.groovy]: Thank you for finding this solution!

**Review by jamesfredley — APPROVED** [review]

**Comment by jamesfredley** [comment]
geb-with-testcontainers should allow Gradle to go back to 8.10.2.

geb-with-webdriver-binaries will be stuck at 8.6, for now.

**Comment by matrei** [comment]
> geb-with-testcontainers should allow Gradle to go back to 8.10.2.
> 
> geb-with-webdriver-binaries will be stuck at 8.6, for now.

@jamesfredley Is it possible to differentiate the gradle version based on selected features?
Do you think we should update to Gradle 8.10.2 even though the optional `geb-with-webdriver-binaries` won't work without manually downgrading created apps?

**Pull request merged**